signer/signer.go

```diff
@@ -1,3 +1,103 @@
 package main
 
-// Write code here
+import (
+	"sort"
+	"strconv"
+	"strings"
+	"sync"
+)
+
+const MaxTh = 6
+var md5Mutex sync.Mutex
+var Memoize sync.Map
+func ExecutePipeline(jobs ...job) {
+	chans := make([]chan interface{}, len(jobs)+1)
+	for i := range chans {
+		chans[i] = make(chan interface{}, 10)
+	}
+	var wg sync.WaitGroup
+	wg.Add(len(jobs))
+	for i, v := range jobs {
+		go worker(chans[i], chans[i+1], v, &wg)
+	}
+	wg.Wait()
+}
+
+func worker(in, out chan interface{}, j job, wg *sync.WaitGroup) {
+	j(in, out)
+	wg.Done()
+	close(out)
```

Comment:
```suggestion
	defer close(out)
```
Так будет лучше и логичнее